eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/tagwrite/BufferedTagWriteManager.java

```diff
@@ -0,0 +1,432 @@
+   * write to the buffered tag write queue, and schedule the buffered tag write processing thread to
+   * process the queue.
+   *
+   * @param managedTagProvider the managed tag provider where the tag is located
+   * @param connectorSettings the connector settings
+   * @param m2WebEwon the M2Web Ewon object for the Ewon where the tag is located
+   * @param m2WebEwonEBDTag the M2Web Ewon EBD tag object for the tag to write
+   * @param tagPath the managed tag provider tag path
+   * @param tagValue the tag value to write
+   * @since 1.0.0
+   */
+  private static void writeTagBuffered(
+      ManagedTagProvider managedTagProvider,
+      EwonConnectorSettings connectorSettings,
+      M2WebEwon m2WebEwon,
+      M2WebEwonEBDTag m2WebEwonEBDTag,
+      TagPath tagPath,
+      Object tagValue) {
+
+    // Add buffered tag write to queue and schedule processing
+    synchronized (BUFFERED_TAG_WRITE_QUEUE) {
+      // Add buffered tag write to queue
+      BUFFERED_TAG_WRITE_QUEUE.add(
+          new BufferedTagWrite(m2WebEwon, m2WebEwonEBDTag, tagPath, tagValue));
+
+      // Schedule buffered tag write processing
+      scheduleBufferedWriteProcessing(managedTagProvider, connectorSettings);
+    }
+
+    // Log debug message
+    if (connectorSettings.isDebugEnabled()) {
+      LOGGER.debug(
+          "Tag write scheduled (buffered) for tag [{}] with value [{}].",
+          tagPath.toString(),
+          tagValue.toString());
+    }
+  }
+
+  /**
+   * Schedules the buffered tag write processing thread to process the buffered tag write queue.
+   * This method will only schedule the processing thread if it is not already running. If the
+   * processing thread is already running, this method will do nothing, as the processing thread
+   * will process the updated queue automatically.
+   *
+   * @param managedTagProvider the managed tag provider where the tag is located
+   * @param connectorSettings the connector settings
+   * @since 1.0.0
+   */
+  private static synchronized void scheduleBufferedWriteProcessing(
+      ManagedTagProvider managedTagProvider, EwonConnectorSettings connectorSettings) {
+    // If buffered tag write processing thread is not running, create and start it
+    if (bufferedTagWriteProcessingThread == null || !bufferedTagWriteProcessingThread.isAlive()) {
+      // Create new thread
+      bufferedTagWriteProcessingThread =
+          new Thread(
+              () -> {
+                // Create boolean to track if thread should continue running
+                boolean continueRunning = true;
+
+                // Loop until queue is empty
+                long totalBufferedWriteCount = 0;
+                while (continueRunning) {
+                  // Wait for buffer length
+                  try {
+                    Thread.sleep(writeBufferLengthMilliseconds);
+                  } catch (InterruptedException e) {
+                    LOGGER.error(
```

Comment:
should an interrupt cause the thread to stop running by setting continueRunning to false?

Reply:
Realistically, this standard Java thread should've been migrated to use the Ignition internal execution manager. I'm not sure an interrupt is necessary, though. A simple boolean flag indicating whether or not the connector is 'shutdown' should work.

Will update

Reply:
Updated with the existing thread to add an interrupt upon connector shutdown. I am going to add an issue to migrate this functionality in the future to an Ignition controlled thread.